openstudiocore/resources/model/OpenStudio.idd

```diff
@@ -26411,3 +26411,367 @@ OS:Meter:CustomDecrement,
+       \memo Input EMS program. a program needs a name
+       \memo a description of when it should be called
+       \memo and then lines of program code for EMS Runtime language
+       \min-fields 4
+  A1, \field Handle
+       \type handle
+       \required-field      
+  A2, \field Name
+       \required-field
+       \type alpha
+       \reference ErlProgramCallingManagerNames
+       \note No spaces allowed in names for ErlProgramCallingManagerNames
+  A3, \field EnergyPlus Model Calling Point
+       \type choice
+       \key BeginNewEnvironment
+       \key AfterNewEnvironmentWarmUpIsComplete
+       \key BeginTimestepBeforePredictor
+       \key AfterPredictorBeforeHVACManagers
+       \key AfterPredictorAfterHVACManagers
+       \key InsideHVACSystemIterationLoop
+       \key EndOfZoneTimestepBeforeZoneReporting
+       \key EndOfZoneTimestepAfterZoneReporting
+       \key EndOfSystemTimestepBeforeHVACReporting
+       \key EndOfSystemTimestepAfterHVACReporting
+       \key EndOfZoneSizing
+       \key EndOfSystemSizing
+       \key AfterComponentInputReadIn
+       \key UserDefinedComponentModel
+       \key UnitarySystemSizing
+  A4; \field Program Name
+       \begin-extensible
+       \required-field
+       \type object-list
+       \note no spaces allowed in name
+       \object-list ErlProgramNames
+
+OS:EnergyManagementSystem:Program,
+       \extensible:1 - repeat last field, remembering to remove ; from "inner" fields.
+       \memo This input defines an Erl program
+       \memo Each field after the name is a line of EMS Runtime Language
+       \min-fields 3
+  A1, \field Handle
+       \type handle
+       \required-field       
+  A2, \field Name
+       \required-field
+       \type alpha
+       \reference ErlProgramNames
+       \reference ErlProgramSubroutineNames
+       \note No spaces allowed in names for ErlProgramNames
+  A3; \field Program Line
+       \begin-extensible
+       \type alpha
+       \required-field
+
+OS:EnergyManagementSystem:Subroutine,
+       \extensible:1 - repeat last field, remembering to remove ; from "inner" fields.
+       \memo This input defines an Erl program subroutine
+       \memo Each field after the name is a line of EMS Runtime Language
+       \min-fields 3
+  A1, \field Handle
+       \type handle
+       \required-field     
+  A2, \field Name
+       \required-field
+       \type alpha
+       \reference ErlSubroutineNames
+       \reference ErlProgramSubroutineNames
+       \note No spaces allowed in names for ErlSubroutineNames
+  A3; \field Program Line
+       \begin-extensible
+       \type alpha
+       \required-field
+  
+OS:EnergyManagementSystem:GlobalVariable,
+       \memo Declares Erl variable as having global scope
+       \memo No spaces allowed in names used for Erl variables
+       \min-fields 2
+  A1, \field Handle
+       \type handle
+       \required-field      
+  A2; \field Name
```

Comment:
yeah, we removed the extensible field in the OSM so we can do uid substitution/tracking for each global variable